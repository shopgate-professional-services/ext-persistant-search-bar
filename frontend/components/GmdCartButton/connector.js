import { connect } from 'react-redux';
import { historyPush } from '@shopgate/pwa-common/actions/router';
import { CART_PATH } from '@shopgate/pwa-common-commerce/cart/constants';

/**
 * @param {Function} dispatch The store dispatch method.
 * @return {Object}
 */
const mapDispatchToProps = dispatch => ({
  navigate: () => dispatch(historyPush({ pathname: CART_PATH })),
});

export default connect(null, mapDispatchToProps);
