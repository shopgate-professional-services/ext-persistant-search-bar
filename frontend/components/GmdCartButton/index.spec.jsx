import React from 'react';
import { mount } from 'enzyme';
import { createMockStore } from '@shopgate/pwa-common/store';
import { Provider } from 'react-redux';
import GmdCartButton from './index';

jest.mock('@shopgate/pwa-common/components/Swiper', () => () => (<div>Swiper</div>));
const store = createMockStore();
const mockProps = {
  count: 1,
  naviagate: () => {},
};
describe('<GmdCartButton />', () => {
  it('should render', () => {
    const component = mount((
      <Provider store={store}>
        <GmdCartButton {...mockProps} />
      </Provider>
    ));
    expect(component).toMatchSnapshot();
  });
});
