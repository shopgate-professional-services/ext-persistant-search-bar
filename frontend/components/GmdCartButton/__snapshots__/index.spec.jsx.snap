// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GmdCartButton /> should render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(GMDCartButton)
    count={1}
    naviagate={[Function]}
  >
    <GMDCartButton
      count={1}
      naviagate={[Function]}
      navigate={[Function]}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={250}
        unmountOnExit={false}
      >
        <Portal
          name="app-bar.cart-button.before"
          props={null}
        />
        <Portal
          name="app-bar.cart-button"
          props={null}
        >
          <div
            className={
              Object {
                "data-css-qxlrct": "",
              }
            }
            style={
              Object {
                "width": 56,
              }
            }
          >
            <AppBarIcon
              background="#fa5400"
              badge={[Function]}
              color="inherit"
              icon={[Function]}
              onClick={[Function]}
              testId="CartButton"
            >
              <button
                className={
                  Object {
                    "data-css-z6y5yb": "",
                  }
                }
                data-test-id="CartButton"
                onClick={[Function]}
                style={
                  Object {
                    "background": "#fa5400",
                    "color": "inherit",
                  }
                }
              >
                <CartIcon>
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>"
                    size="inherit"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M7.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2s2-0.9,2-2S9,18,7.9,18z M17.9,18c-1.1,0-2,0.9-2,2s0.9,2,2,2c1.1,0,2-0.9,2-2S19,18,17.9,18z M8.1,14.8l0-0.1L9,13h7.5c0.7,0,1.4-0.4,1.7-1l3.9-7l-1.7-1h0l-1.1,2l-2.8,5h-7l-0.1-0.3L7.1,6L6.1,4L5.2,2H1.9v2h2l3.6,7.6L6.2,14c-0.2,0.3-0.2,0.6-0.2,1c0,1.1,0.9,2,2,2h12v-2H8.3C8.2,15,8.1,14.9,8.1,14.8z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "inherit",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </CartIcon>
                <Component>
                  <CartButtonBadge
                    count={1}
                  >
                    <div
                      className={
                        Object {
                          "data-css-7g35sj": "",
                        }
                      }
                    >
                      1
                    </div>
                  </CartButtonBadge>
                </Component>
              </button>
            </AppBarIcon>
          </div>
        </Portal>
        <Portal
          name="app-bar.cart-button.after"
          props={null}
        />
      </Transition>
    </GMDCartButton>
  </Connect(GMDCartButton)>
</Provider>
`;
